### PR TITLE
If possible, do not use FlagValues.FlagDict.  Or use FlagValues._flags instead.

### PR DESCRIPTION
**Pull request by yilei**
This is a followup for https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/1175/commits/4b6c2dbeac5a2fc2b46686a78fd65da468631ca7

### CODE REVIEW

**Comment by gareth-ferneyhough** [comment]
LGTM. @tedsta, can you check this out too?

**Comment by yilei** [comment]
Kindly ping? @tedsta

File: tests/configs/benchmark_config_spec_test.py [hunk 1]
```diff
@@ -38,6 +38,10 @@
 _GCP_AWS_DISK_CONFIG = {'GCP': {}, 'AWS': {}}
 
 
+def _GetFlagDict(flag_values):
+  return {name: flag_values[name] for name in flag_values}
```
Comment (tedsta) [inline tests/configs/benchmark_config_spec_test.py]: IIUC, this builds an actual dict out of something that emulates a dict?

File: tests/flag_util_test.py [hunk 2]
```diff
@@ -120,7 +120,11 @@ def testReadAndWrite(self):
     flag_values_copy.test_flag = 1
     self.assertFlagState(flag_values, 0, False)
     self.assertFlagState(flag_values_copy, 1, False)
-    with flag_util.FlagDictSubstitution(flag_values, flag_values_copy.FlagDict):
+    if hasattr(flag_values_copy, '_flags'):
+      flag_dict_func = flag_values_copy._flags
+    else:
+      flag_dict_func = flag_values_copy.FlagDict
```
Comment (tedsta) [inline tests/flag_util_test.py]: Interesting that there's conditionals in a unit test. Shouldn't there only ever be one path through a unit test?

File: tests/configs/benchmark_config_spec_test.py (hunk 1 above)
Comment (yilei) [inline tests/configs/benchmark_config_spec_test.py]: The flags.FLAGS object is a dict-like object, the tests is comparing a dictionary with _GetFlagDict().

File: tests/flag_util_test.py (hunk 2 above)
Comment (yilei) [inline tests/flag_util_test.py]: FlagDict is going away internally, and unfortunately _flags is not available in github.com/google/python-gflags (yet), and the test needs to run in both environments.

I'm going to bring the two flags library to a same state, but it takes sometime, that results in this ugly intermediate state on your code :(

**Pull request merged**